packages/amplify-category-auth/src/provider-utils/awscloudformation/service-walkthrough-types.ts

```diff
@@ -0,0 +1,100 @@
+// Some convenience types for the existing service walkthrough logic
+
+export type ServiceQuestionsResult = ServiceQuestionsBaseResult &
+  OAuthResult &
+  SocialProviderResult &
+  IdentityPoolResult &
+  PasswordPolicyResult &
+  PasswordRecoveryResult &
+  MfaResult &
+  AdminQueriesResult;
+
+export interface ServiceQuestionsBaseResult {
+  serviceName: string;
+  resourceName?: string;
+  useDefault: 'manual';
+  requiredAttributes: string[];
+  authSelections: 'userPoolOnly' | 'identityPoolAndUserPool';
+  userPoolName?: string;
+  usernameAttributes?: UsernameAttributes;
+  userPoolGroups: boolean;
+  userPoolGroupList?: string[];
+  userpoolClientRefreshTokenValidity?: number;
+  userpoolClientReadAttributes: string[];
+  userpoolClientWriteAttributes: string[];
+}
+
+export interface OAuthResult {
+  hostedUI: boolean;
+  hostedUIDomainName?: string;
+  hostedUIProviderMeta?: any;
+  hostedUIProviderCreds?: any;
+  AllowedOAuthFlows?: 'code' | 'implicit';
```

Comment:
Why is it uppercase and the rest is not? it is coming from existing code?

Reply:
Yeah this is a type I made for existing code